docs/DesignDocs/SGX_EMM.md

```diff
@@ -0,0 +1,350 @@
+* @param[in] length size of the region in multiples of page size in bytes
+* @param[in] flags a bitwise OR of flags describing committing mode, committing order, address preference.
+*        Flags should include exactly one of following for committing mode:
+*            - EMA_RESERVE: just reserve an address range, no EPC commited. To allocate memory on a reserved range, 
+*                           call this function again with EMA_COMMIT_ON_DEMAND or EMA_COMMIT_NOW.
+*            - EMA_COMMIT_NOW: reserves memory range and commit EPC pages. EACCEPT for all allocated pages 
+*                              are done when this function returns.
+*            - EMA_COMMIT_ON_DEMAND: reserves memory range, EPC pages are committed (EACCEPT) on demand upon #PF.
+*        ORed with zero or one of the committing order flags:
+*            - EMA_GROWSDOWN: always commit pages from higher to lower addresses, no gaps in addresses above the last commited
+*            - EMA_GROWSUP:   always commit pages from lower to higher addresses, no gaps in addresses below the last commited.
+*        Optionally ORed with EMA_FIXED to indicate allocation at fixed address. 
+*
+* @param[in] handler, custom handler for page faults in this region, NULL if no custom handling needed
+* @param[in] handler_private, private data for the @handler, which will be passed back when the handler is called.
+* @param[out] out_addr, pointer to store the start address of allocated range. Set to valid address by the function on success, NULL otherwise. 
+* @return 0 on success. EEXIST if range requested is overlapping a region in use and EMA_FIXED is requested,
+* ENOMEM for out of memory, EACCES if region is outside enclave address space.
+*/
+int sgx_mm_alloc(void *addr, size_t length, int flags, enclave_fault_handler_t handler, void* handler_private, void** out_addr)
+```
+
+**Remarks:**
+- If the desired address/range is in use, return a different start address if EMA_FIXED flag is not set.
+- Permissions of newly allocated regions are always PROT_READ|PROT_WRiTE and page type PT_REG, except for EMA_RESERVE mode regions which will have PROT_NONE.
+- Once allocated by sgx_mm_alloc, a region will stay in allocated state and become deallocated once sgx_mm_dealloc is called. 
+- If sgx_mm_dealloc on a partial range of a previously allocated region, then the region is split,and the freed range is deallocated. The remainder of the region stays allocated. 
+- If all pages in the region are freed by sgx_mm_dealloc, then the whole region is released, and the memory manager no longer tracks the region. 
+
+
+### sgx_mm_uncommit and sgx_mm_dealloc
+
+```
+/**
+* Uncommit (trim) physical EPC pages in a previously committed range. 
+* The pages in the allocation are freed, but the address range is still reserved.
+* @param[in] addr page aligned start address of the region to be trimmed
+* @param[in] length size in bytes of multiples of page size.
```

Comment:
nit: be consistent in terms of whether argument descriptions in in period or not